src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -130,23 +116,27 @@ export class KubernetesCluster extends CatalogEntity<CatalogEntityMetadata, Kube
       );
     }
 
-    if (this.status.phase == "connected") {
-      context.menuItems.push({
-        title: "Disconnect",
-        icon: "link_off",
-        onClick: () => requestMain(clusterDisconnectHandler, this.metadata.uid)
-      });
-    } else {
-      context.menuItems.push({
-        title: "Connect",
-        icon: "link",
-        onClick: () => context.navigate(`/cluster/${this.metadata.uid}`)
-      });
+    switch (this.status.phase) {
+      case "connected":
+      case "connecting":
+        context.menuItems.push({
+          title: "Disconnect",
+          icon: "link_off",
+          onClick: () => requestMain(clusterDisconnectHandler, this.metadata.uid)
+        });
+        break;
+      case "disconnected":
+        context.menuItems.push({
+          title: "Connect",
+          icon: "link",
+          onClick: () => context.navigate(`/cluster/${this.metadata.uid}`)
+        });
+        break;
     }
 
-    const category = catalogCategoryRegistry.getCategoryForEntity<KubernetesClusterCategory>(this);
-
-    if (category) category.emit("contextMenuOpen", this, context);
+    catalogCategoryRegistry
```

Comment:
I'm not really a fan of these changes. Both are totally valid code and the change is not necessary, they are just causing noise. Or did I miss something?

Reply:
Do you specifically mean the change to how the emit is handled or even the switch?

Reply:
I mean that isn't you change doing exactly the same as
```
const category = catalogCategoryRegistry.getCategoryForEntity<KubernetesClusterCategory>(this);

if (category) category.emit("contextMenuOpen", this, context);
```